src/messages.rs

```diff
@@ -68,6 +73,62 @@ impl From<Message> for Vec<u8> {
     }
 }
 
+impl TryFrom<(libp2p::gossipsub::Message, pchain_types::cryptography::PublicAddress)> for Message {
+    type Error = MessageConversionError;
+
+    fn try_from((message , local_public_address): (libp2p::gossipsub::Message, pchain_types::cryptography::PublicAddress)) 
+    -> Result<Self, Self::Error> {
+        let (topic_hash, data) = (message.topic, message.data);
+        let mut data = data.as_slice();
+        
+        let topic = identify_topics(topic_hash, local_public_address)?;
+        
+        match topic {
+            HotStuffRsBroadcast | HotStuffRsSend(_) => {
+                let message = hotstuff_rs::messages::Message::deserialize(&mut data).map(Message::HotStuffRs)?;
+                Ok(message)
+            },
+            Mempool => {
+                let message = pchain_types::blockchain::TransactionV1::deserialize(&mut data).map(Message::Mempool)?;
+                Ok(message)
+            },
+            DroppedTxns => {
+                let message = DroppedTxnMessage::deserialize(&mut data).map(Message::DroppedTxns)?;
+                Ok(message)
+            }
+        }
+    }
+}
+
+fn identify_topics(topic_hash: TopicHash, addr: PublicAddress) -> Result<Topic, InvalidTopicError> {
```

Comment:
Don't need a separate function for identifying the topics. I think it can be put into the TryFrom trait directly. Make the TryFrom trait take in the TopicHash too.
Even if you wanna keep it separated, have to change the name, as they mentioned the name "identify_xxxx" is overlapping with the libp2p module